courseme/static/js/edit-objective-modal.js

```diff
@@ -49,44 +46,30 @@ $(document).ready(function () {
         //console.log(jQuery.isPlainObject( data_json ));
         //console.log(typeof data_json);        
 
-        
-        
-        $.post(  
-            flask_util.url_for('main.objective_add_update'),
-            data,  
-            function(json) {
-                
-                console.log(json);
-                
-                var result = $.parseJSON(json);
-                console.log(result);
-                console.log(result["savedsuccess"]);
-                console.log(result.savedsuccess);                
-                
-                $("#edit_objective_form").find(".help-block").text("");
-                            
-                if(result.savedsuccess)
-                {
-                    $("#edit_objective_modal").modal('hide');
-                    location.reload();
+
+        $.post(flask_util.url_for('main.objective_add_update'), data)
+            .done(function(response) {  // on success
```

Comment:
Using the Promise-like API usually makes things a bite more manageable in JS.  Doesn't make much difference in this case (since we're not chaining requests), but it's good to be aware of it if you're not already.